File: app/src/main/java/com/example/speedrun/injection/modules/ActivityModule.java
package com.example.speedrun.injection.modules;


import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {

    private Activity activity;

    public ActivityModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    Activity provideActivity() {
        return activity;
    }

    @Provides
    Context provideContext() {
        return activity;
    }

}

File: app/src/main/java/com/example/speedrun/injection/modules/ActivityModule.kt
package com.example.speedrun.injection.modules


import android.app.Activity
import android.content.Context

import dagger.Module
import dagger.Provides

@Module
class ActivityModule(private val activity: Activity) {

    @Provides
    fun provideActivity(): Activity {
        return activity
    }

    @Provides
    fun provideContext(): Context {
        return activity
    }

}
